dj/models/node.py

```diff
@@ -195,17 +209,41 @@ class NodeAvailabilityState(SQLModel, table=True):  # type: ignore
     )
 
 
-class Node(NodeBase, table=True):  # type: ignore
+class ReferenceNode(ReferenceNodeBase, table=True):  # type: ignore
     """
-    A node.
+    Reference node that acts as an umbrella for all node revisions
     """
 
     id: Optional[int] = Field(default=None, primary_key=True)
-    status: NodeStatus = NodeStatus.INVALID
+
+    current_version: int = Field(default=1)
     created_at: UTCDatetime = Field(
         sa_column=SqlaColumn(DateTime(timezone=True)),
         default_factory=partial(datetime.now, timezone.utc),
     )
+
+    revisions: List["Node"] = Relationship(back_populates="reference_node")
+    current: "Node" = Relationship(
+        sa_relationship_kwargs={
+            "primaryjoin": "and_(ReferenceNode.id==Node.reference_node_id, "
+            "ReferenceNode.current_version == Node.version)",
+            "viewonly": True,
+            "uselist": False,
+        },
+    )
+
+
+class Node(NodeBase, table=True):  # type: ignore
```

Comment:
A naming suggestion: Since `ReferenceNode` will be the common entry point that's used most around the codebase, why not rename that to `Node` and then rename `Node` here to `Revision`? That way `Node.revisions` gives you `List[Revision]`.

Reply:
+1. I think what @samredai suggests is more intuitive. One adjustment to what he said could be `s/Revision/NodeRevision`. This way the class name on its own is also pretty informative.

Reply:
@agorajek and @samredai  -- agreed on naming changes. I went ahead and did `ReferenceNode` -> `Node` and `Node` -> `NodeRevision`.